tmail-backend/jmap/extensions-postgres/src/main/java/com/linagora/tmail/james/jmap/settings/PostgresJmapSettingsDAO.java

```diff
@@ -0,0 +1,129 @@
+import org.apache.james.core.Domain;
+import org.apache.james.core.Username;
+import org.apache.james.jmap.core.UuidState;
+import org.jooq.Record;
+import org.jooq.impl.DSL;
+import org.jooq.postgres.extensions.types.Hstore;
+
+import com.google.common.collect.ImmutableMap;
+
+import reactor.core.publisher.Mono;
+import scala.jdk.javaapi.CollectionConverters;
+
+public class PostgresJmapSettingsDAO {
+    public static class Factory {
+        private final PostgresExecutor.Factory executorFactory;
+
+        @Inject
+        @Singleton
+        public Factory(PostgresExecutor.Factory executorFactory) {
+            this.executorFactory = executorFactory;
+        }
+
+        public PostgresJmapSettingsDAO create(Optional<Domain> domain) {
+            return new PostgresJmapSettingsDAO(executorFactory.create(domain));
+        }
+    }
+
+    private final PostgresExecutor postgresExecutor;
+
+    public PostgresJmapSettingsDAO(PostgresExecutor postgresExecutor) {
+        this.postgresExecutor = postgresExecutor;
+    }
+
+    public Mono<JmapSettings> getJmapSettings(Username username) {
+        return postgresExecutor.executeRow(dslContext -> Mono.from(dslContext.selectFrom(TABLE_NAME)
+            .where(USER.eq(username.asString()))))
+            .map(this::toJmapSettings);
+    }
+
+    public Mono<UuidState> getState(Username username) {
+        return postgresExecutor.executeRow(dslContext -> Mono.from(dslContext.select(STATE)
+                .from(TABLE_NAME)
+                .where(USER.eq(username.asString()))))
+            .map(record -> new UuidState(record.get(STATE)));
+    }
+
+    public Mono<Void> saveSettings(Username username, UuidState newState, Map<JmapSettingsKey, JmapSettingsValue> newSettings) {
+        return postgresExecutor.executeVoid(dslContext -> Mono.from(dslContext.insertInto(TABLE_NAME)
+            .set(USER, username.asString())
+            .set(STATE, newState.value())
+            .set(SETTINGS, toHstore(newSettings))));
+    }
+
+    public Mono<Void> updateFullSettings(Username username, UuidState newState, Map<JmapSettingsKey, JmapSettingsValue> newSettings) {
+        return postgresExecutor.executeVoid(dslContext -> Mono.from(dslContext.update(TABLE_NAME)
+            .set(STATE, newState.value())
+            .set(SETTINGS, toHstore(newSettings))
+            .where(USER.eq(username.asString()))));
+    }
+
+    public Mono<Void> updateSettings(Username username,
+                                     UuidState newState,
+                                     Map<JmapSettingsKey, JmapSettingsValue> addSettings,
+                                     List<JmapSettingsKey> removeSettings) {
+        return postgresExecutor.executeVoid(dslContext -> Mono.from(dslContext.update(TABLE_NAME)
+            .set(STATE, newState.value())
+            .set(DSL.field(SETTINGS.getName()),
+                (Object) DSL.field("delete(" + SETTINGS.getName() + " || ?, ?)",
```

Comment:
This string concatenation in a sql query is more than likely to be prone to a SQL injection.

Reply:
Don't worry. SETTINGS.getName() is constant